lib/gpu/lib/src/render_pass.dart

```diff
@@ -150,6 +150,18 @@ base class SamplerOptions {
   SamplerAddressMode heightAddressMode;
 }
 
+base class Scissor {
+  Scissor({this.x = 0, this.y = 0, this.width = 0, this.height = 0});
+
+  int x, y, width, height;
+
+  void _validate() {
+    if (x < 0 || y < 0 || width < 0 || height < 0) {
+      throw Exception("Invalid values for scissor. All values should be positive.");
```

Comment:
Ah, one more snag for this, actually: Could you please add a test that ensures this validation fires?
There are a bunch of other tests around which check exceptions to reference.

Reply:
Done

Reply:
It doesn't look like this comment was addressed?

Reply:
Sorry about that,  changes got missed while pushing. All the changes are pushed now.